### DISCUSSION TITLE
16 bit interface

### DISCUSSION THREAD

**Discussion by kpetrovich** [Bodmer/TFT_eSPI#1553] [Q&A - General] [answered]
Hi. Is it possible to rewrite the library for a 16-bit parallel interface for esp32. theoretically, esp32 has enough pins ?

**Comment by Bodmer** [comment, marked as answer]
Yes, that is possible.

**Reply by kpetrovich** [reply]
how about this https://github.com/Bodmer/TFT_eSPI/discussions/1555

you wrote that the esp 32 can only work with an 8 bit bus

**Reply by Bodmer** [reply]
Yes, the library is compatible with an 8 bit parallel interface to the TFT.  Adapting the code for 16 bit would be possible but that is not something I am planning to do.

**Reply by udem1234** [reply]
Hi Bodmer.

We are using MakerFabs "MaTouch_ESP32-S3 Parallel TFT with Touch 3.5'' ILI9488" board that use a 16-bit parallel interface for the TFT.  I've read lot of positive performance comparison about your library and I would prefer it rather than LGFX (LovyanGFX).  However, 16-bit support is essential.

Are you still not planning to support 16-bit ?

**Reply by Bodmer** [reply]
16 bit is only supported with the RP2040 processor. There are no plans to add 16 bit support to other processors at the moment.
